astropy/utils/decorators.py

```diff
@@ -936,3 +936,93 @@ def set_docstring(obj):
             obj.__doc__ = doc.format(*args, **kwargs)
         return obj
     return set_docstring
+
+
+def replaced_parameter(oldkeyword, newkeyword, since):
+    """Enable a function with a new keyword signature, to still use the
+    deprecated old keyword.
+
+    The decorated function should follow the new signature: the
+    definition should use ``newkeyword`` instead of ``oldkeyword``.
+    The new parameter should behave the same as the old parameter.
+
+    An AstropyDeprecationWarning is given if ``oldkeyword`` is used,
+    and a TypeError is raised if both ``oldkeyword`` and
+    ``newkeyword`` are used in the decorated function.
+
+    Parameters
+    ----------
+
+    oldkeyword : str
+        The keyword in the function signature that has been replaced
+
+    newkeyword : str
+        The keyword in the function signature replaces oldkeyword
+
+    since : str
+        The release when this change was made.
+
+
+    Raises
+    ------
+
+    TypeError
+        If both ``newkeyword`` and ``oldkeyword`` are used.
+
+    Example
+    -------
+
+    This decorator originated when replacing the `clobber` parameter
+    with `overwrite`. For example, a function::
+
+        def test(a, b=10, clobber=False):
+            ...
+
+    becomes, when replacing clobber with overwrite::
+
+        @replaced_parameter('clobber', 'overwrite', since='1.3')
+        def test(a, b=10, overwrite=False):
+            ...
+
+    """
+    def decorator(function):
+        # Avoid circular import
+        from .compat.funcsigs import signature
+        # We update the doc string (if it exists), *before* functools.wraps
+        if function.__doc__:
+            function.__doc__ += ("\n.. versionchanged:: {version}"
+                                 "\n   ``{new}`` replaces the deprecated "
+                                 "``{old}`` keyword\n".format(
+                                 version=since, new=newkeyword,
+                                 old=oldkeyword))
+        @functools.wraps(function)
+        def wrapper(*args, **kwargs):
+            # Determine the position of the possible newkeyword
+            # positional parameter
+            posargs = list(signature(function).parameters.keys())
```

Comment:
as mentioned in the other PR you don't need to do this every time the function is called but can be done within the ``decorator``-function.